libs/ledger/include/ledger/shards/service_identifier.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/serializers/group_definitions.hpp"
+
+#include <cstdint>
+#include <string>
+
+namespace fetch {
+namespace ledger {
+
+class ServiceIdentifier
```

Comment:
how is this used?

Reply:
It is the key in the manifest table used to identity a services like a lane. In the future we might refactor this subsystem a little. This is mostly a port of the code from the old p2p subsystem.